### PR TITLE
Doctest and typing for longest_increasing_subsequence.py

### PR DESCRIPTION
**Pull request by poyea**
* Type hint for the algorithm
* Doctests added

### CODE REVIEW

File: dynamic_programming/longest_increasing_subsequence.py [hunk 1]
```diff
@@ -7,10 +7,21 @@
 Given an ARRAY, to find the longest and increasing sub ARRAY in that given ARRAY and return it.
 Example: [10, 22, 9, 33, 21, 50, 41, 60, 80] as input will return [10, 22, 33, 41, 60, 80] as output
 """
-
-
-def longestSub(ARRAY):  # This function is recursive
```
Comment (cclauss) [inline dynamic_programming/longest_increasing_subsequence.py]: I gotta say that I am not a fan of all these UPPERCASE VARIABLE NAMES as it runs counter to
https://pep8.org/#prescriptive-naming-conventions

File: dynamic_programming/longest_increasing_subsequence.py [hunk 2]
```diff
@@ -4,50 +4,50 @@
...
 The problem is  :
-Given an ARRAY, to find the longest and increasing sub ARRAY in that given ARRAY and return it.
+Given an array, to find the longest and increasing sub ARRAY in that given array and return it.
 Example: [10, 22, 9, 33, 21, 50, 41, 60, 80] as input will return [10, 22, 33, 41, 60, 80] as output
 """
 from typing import List
 
 
-def longestSub(ARRAY: List[int]) -> List[int]:  # This function is recursive
+def longest_subsequence(Array: List[int]) -> List[int]:  # This function is recursive
```
Comment (cclauss) [inline dynamic_programming/longest_increasing_subsequence.py]: Array --> array please.  Please see CONTRIBUTING.md for info on variable naming in Python.

File: dynamic_programming/longest_increasing_subsequence.py [hunk 3]
```diff
@@ -4,50 +4,50 @@
...
     [1, 2, 3, 9]
-    >>> longestSub([9, 8, 7, 6, 5, 7])
+    >>> longest_subsequence([9, 8, 7, 6, 5, 7])
     [8]
-    >>> longestSub([1, 1, 1])
+    >>> longest_subsequence([1, 1, 1])
     [1, 1, 1]
     """
-    ARRAY_LENGTH = len(ARRAY)
+    ArrayLength = len(Array)
```
Comment (cclauss) [inline dynamic_programming/longest_increasing_subsequence.py]: array_length please

File: dynamic_programming/longest_increasing_subsequence.py [hunk 4]
```diff
@@ -4,50 +4,50 @@
...
     if (
-        ARRAY_LENGTH <= 1
+        ArrayLength <= 1
     ):  # If the array contains only one element, we return it (it's the stop condition of recursion)
-        return ARRAY
+        return Array
         # Else
-    PIVOT = ARRAY[0]
+    Pivot = Array[0]
```
Comment (cclauss) [inline dynamic_programming/longest_increasing_subsequence.py]: pivot please

File: dynamic_programming/longest_increasing_subsequence.py [hunk 5]
```diff
@@ -4,50 +4,50 @@
...
     isFound = False
     i = 1
-    LONGEST_SUB = []
-    while not isFound and i < ARRAY_LENGTH:
-        if ARRAY[i] < PIVOT:
+    LongestSubseq = []
```
Comment (cclauss) [inline dynamic_programming/longest_increasing_subsequence.py]: longest_subseq please

File: dynamic_programming/longest_increasing_subsequence.py [hunk 6]
```diff
@@ -4,50 +4,50 @@
...
+    while not isFound and i < ArrayLength:
+        if Array[i] < Pivot:
             isFound = True
-            TEMPORARY_ARRAY = [element for element in ARRAY[i:] if element >= ARRAY[i]]
-            TEMPORARY_ARRAY = longestSub(TEMPORARY_ARRAY)
-            if len(TEMPORARY_ARRAY) > len(LONGEST_SUB):
-                LONGEST_SUB = TEMPORARY_ARRAY
+            TempArray = [element for element in Array[i:] if element >= Array[i]]
```
Comment (cclauss) [inline dynamic_programming/longest_increasing_subsequence.py]: temp_array

File: dynamic_programming/longest_increasing_subsequence.py (hunk 2 above)
Comment (poyea) [reply dynamic_programming/longest_increasing_subsequence.py]: Woo, my bad. Followed strictly the PEP8.

**Pull request merged**